### PR TITLE
[Kettle] Limit days of builds to `all` table to avoid deadlock

### PR DESCRIPTION
**Pull request by MushuEE**
Due to #20024 we are seeing kettle prod hand indefinitely on `make_json` for the `all` table. This seems to not be an issue if the number of builds collected and pushed is limited to 30 days. (Hung on 90+ days).

30 days will back update kettle to before the `all` table stopped being populated and will limit the collection time. This does not write over existing older builds and will be safe for BQ. This should be reverted when there is a more permanent solution or a change in architecture.

This change was tested in Kettle and successfully updated `all`.

I also added some logging to better catch the odd exception I had seen in #19732

/cc @amwat @spiffxp @roberthbailey 
/area Kettle

### CODE REVIEW

**Comment by MushuEE** [comment]
/hold this might be counting dupes without including the --replace. (We only want to replace data from the last 30 days)

- Can we break the make_json into multiple X day blocks?

**Comment by MushuEE** [comment]
UPDATE: The way kettle works, it creates a unique table based on the days specified. Previously there was no 30 day table. This caused builds that had been uploaded to be double counted in the `build.all` 
`SELECT * FROM k8s-gubernator.build.all WHERE started between '2020-11-1 22:52:14 UTC' and '2020-11-8 22:52:14 UTC' order by number LIMIT 1000`

This should not repeat going forward but will cause an issue if reverting back to `all` and not `--30` I can remove duplicates however (i believe) @spiffxp 

/hold cancel

**Review by spiffxp — CHANGES_REQUESTED** [review]
Can you rebase to avoid the merge commits?

File: kettle/Dockerfile [hunk 1]
```diff
@@ -28,6 +28,7 @@ RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime \
     python \
     python3 \
     python3-pip \
+    vim \
```
Comment (amwat) [inline kettle/Dockerfile:31]: we usually avoid adding text editors to prod images, i'm guessing this was needed during debugging?

**Review by amwat — COMMENTED** [review]
/lgtm
/hold

feel free to cancel the hold if the duplicate data issue is resolved.

File: kettle/Dockerfile (hunk 1 above)
Comment (MushuEE) [reply kettle/Dockerfile:31]: Yes it is just generally helpful but I can remove

**Comment by MushuEE** [comment]
@spiffxp 30 day issue will not repeat duplicates but I would like to get this in soon if possible. I am tempted to backup the older database and either clear it or start fresh with a new one to hopefully give kettle back some life.

**Comment by amwat** [comment]
/lgtm
/hold

@MushuEE Is this good to merge?
feel free to cancel the hold if all the issues are resolved.

**Comment by spiffxp** [comment]
The triage job has started failing because it's getting zero results, I'm assuming merging/deploying this might bring it back to green (though we should fix triage code to handle zero results) https://github.com/kubernetes/test-infra/issues/20259

**Pull request merged**